sycl/include/sycl/ext/intel/experimental/esimd/memory.hpp

```diff
@@ -613,20 +613,27 @@ lsc_gather(AccessorTy acc, __ESIMD_NS::simd<uint32_t, N> offsets,
 ///
 template <typename T, int NElts, lsc_data_size DS = lsc_data_size::default_size,
           cache_hint L1H = cache_hint::none, cache_hint L3H = cache_hint::none,
-          typename FlagsT = __ESIMD_NS::element_aligned_tag,
-          typename = std::enable_if_t<__ESIMD_NS::is_simd_flag_type_v<FlagsT>>>
-__ESIMD_API __ESIMD_NS::simd<T, NElts>
+          typename FlagsT = __ESIMD_NS::overaligned_tag<4>>
```

Comment:
Gregory, I believe what we agreed on the default value previously was: std::max(__ESIMD_NS::overaligned_tag<4>, __ESIMD_NS::element_aligned_tag).
So, passing int64_t pointer would assume element_aligned_tag (i.e. 8-byte alignment) and would not require explicit passing of the tag argument for longs.

From other side, if user explicitly passes overaligned<4> for 'int64_t* ptr', then that may be an indicator of
some alignment problems in users's code and it would be good to report that in compile time.
The latest update of this PR handles this scenario differently,
the explicitly set alignment to 4 would be just ignored.

It is not possible to write std::max as shown above, but we can have something like 'dqword_element_aligned_tag' defined in detail namespace, which would return 4 for byte/short/int and 8 for int64_t.
That also would remove the need in Alignment evaluation on the lines 625-632

That is a temporary solution before we get approval for using element_aligned_tag that breaks backward compatibility with previous calls with int8_t* and int16_t* without explicit alignment tag passed.